<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Set of packages for doing trajectory inference on single-cell data • dyno</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Set of packages for doing trajectory inference on single-cell data">
<meta property="og:description" content="The dyno package automatically loads in packages to:
  select the best TI methods for a given problem (dynguidelines)
  run these methods locally or in a docker environment (dynmethods)
  plot the predictions (dynplot)
  select relevant features (dynfeature)
  transform and work with the trajectory (dynwrap)">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">dyno</a>
        <span class="label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/dyno.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dynverse/dyno">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    

    
    
<!-- README.md is generated from README.Rmd. Please edit that file -->
<p><a href="https://travis-ci.org/dynverse/dyno"><img src="https://img.shields.io/travis/dynverse/dyno.svg?logo=travis" alt="Travis"></a> <a href="https://ci.appveyor.com/project/dynverse/dyno"><img src="https://ci.appveyor.com/api/projects/status/github/dynverse/dyno?branch=master&amp;svg=true" alt="AppVeyor Build Status"></a> <img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle"></p>
<p>The <strong>dyno</strong> package offers <strong>end-users</strong> a complete TI pipeline. It features:</p>
<ul>
<li>a uniform interface to 57 <a href="https://github.com/dynverse/dynmethods#list-of-included-methods">TI methods</a>,</li>
<li>an <a href="https://github.com/dynverse/dyno#selecting-the-most-optimal-ti-methods">interactive guideline tool</a> to help the user select the most appropriate method,</li>
<li>
<a href="https://github.com/dynverse/dyno#plotting-the-trajectory">streamlined interpretation and visualisation of trajectories</a>, including colouring by gene expression or clusters, and</li>
<li>downstream analyses such as the <a href="https://github.com/dynverse/dyno#predicting-and-visualising-genes-of-interest">identification of potential marker genes</a>.</li>
</ul>
<p>For information on how to install dyno, check out the <a href="https://github.com/dynverse/dyno#installation">installation instructions below</a>.</p>
<div id="trajectory-inference-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#trajectory-inference-workflow" class="anchor"></a>Trajectory inference workflow</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dyno)
<span class="kw">library</span>(tidyverse)</code></pre></div>
<p>The whole trajectory inference workflow is divided in several steps:</p>
<div class="figure">
<img src="reference/figures/toolkit.png">
</div>
<div id="preparing-the-data" class="section level3">
<h3 class="hasAnchor">
<a href="#preparing-the-data" class="anchor"></a>Preparing the data</h3>
<p>The first step is to prepare the data for trajectory inference using <code>wrap_expression</code>. It requires both the counts and normalised expression (with genes/features in columns) as some TI methods are specifically built for one or the other:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"fibroblast_reprogramming_treutlein"</span>)

dataset &lt;-<span class="st"> </span><span class="kw"><a href="reference/wrap_expression.html">wrap_expression</a></span>(
  <span class="dt">counts =</span> fibroblast_reprogramming_treutlein<span class="op">$</span>counts,
  <span class="dt">expression =</span> fibroblast_reprogramming_treutlein<span class="op">$</span>expression
)</code></pre></div>
</div>
<div id="selecting-the-most-optimal-ti-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#selecting-the-most-optimal-ti-methods" class="anchor"></a>Selecting the most optimal TI methods</h3>
<p>When the data is wrapped, the most performant and scalable set of tools can be selected using a shiny app. This app will select a set of methods which are predicted to produce the most optimal output given several user-dependent factors (such as prior expectations about the topology present in the data) and dataset-dependent factors (such as the size of the dataset). This app uses the benchmarking results from <a href="https://github.com/dynverse/dynbenchmark">dynbenchmark</a> (<a href="https://doi.org/10.1101/276907">doi:10.1101/276907</a>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">guidelines &lt;-<span class="st"> </span><span class="kw"><a href="reference/guidelines.html">guidelines_shiny</a></span>(dataset)
methods &lt;-<span class="st"> </span>guidelines<span class="op">$</span>methods <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(selected) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(method_id) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">first</span>()</code></pre></div>
<div class="figure">
<img src="reference/figures/dynguidelines.gif">
</div>
</div>
<div id="running-the-methods" class="section level3">
<h3 class="hasAnchor">
<a href="#running-the-methods" class="anchor"></a>Running the methods</h3>
<p>We wrapped available methods were wrapped within a common interface, which makes running a method a one-step-process. For running the methods, it is currently necessary to <a href="#Installation">have docker installed</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model &lt;-<span class="st"> </span><span class="kw"><a href="reference/infer_trajectories.html">infer_trajectory</a></span>(dataset, <span class="kw">first</span>(methods))</code></pre></div>
</div>
<div id="interpreting-the-trajectory-biologically" class="section level3">
<h3 class="hasAnchor">
<a href="#interpreting-the-trajectory-biologically" class="anchor"></a>Interpreting the trajectory biologically</h3>
<p>In most cases, some knowledge is present of the different start, end or intermediary states present in the data, and this can be used to adapt the trajectory so that it is easier to interpret.</p>
<div id="rooting" class="section level4">
<h4 class="hasAnchor">
<a href="#rooting" class="anchor"></a>Rooting</h4>
<p>Most methods have no direct way of inferring the directionality of the trajectory. In this case, the trajectory should be “rooted” using some external information, for example by using a set of marker genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model &lt;-<span class="st"> </span>model <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw"><a href="reference/add_root_using_expression.html">add_root_using_expression</a></span>(<span class="kw">c</span>(<span class="st">"Vim"</span>), dataset<span class="op">$</span>expression)</code></pre></div>
</div>
<div id="milestone-labelling" class="section level4">
<h4 class="hasAnchor">
<a href="#milestone-labelling" class="anchor"></a>Milestone labelling</h4>
<p>Milestones can be labelled using marker genes. These labels can then be used for subsequent analyses and for visualisation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model &lt;-<span class="st"> </span><span class="kw"><a href="reference/label_milestones.html">label_milestones_markers</a></span>(
  model,
  <span class="dt">markers =</span> <span class="kw">list</span>(
    <span class="dt">MEF =</span> <span class="kw">c</span>(<span class="st">"Vim"</span>),
    <span class="dt">Myocyte =</span> <span class="kw">c</span>(<span class="st">"Myl1"</span>),
    <span class="dt">Neuron =</span> <span class="kw">c</span>(<span class="st">"Stmn3"</span>)
  ),
  dataset<span class="op">$</span>expression
)</code></pre></div>
</div>
</div>
<div id="plotting-the-trajectory" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-the-trajectory" class="anchor"></a>Plotting the trajectory</h3>
<p>Several visualisation methods provide ways to biologically interpret trajectories.</p>
<p>Examples include combining a dimensionality reduction, a trajectory model and a cell clustering:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model &lt;-<span class="st"> </span>model <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="reference/add_dimred.html">add_dimred</a></span>(dyndimred<span class="op">::</span>dimred_mds, <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression)
<span class="kw"><a href="reference/plot_dimred.html">plot_dimred</a></span>(
  model, 
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression, 
  <span class="dt">grouping =</span> fibroblast_reprogramming_treutlein<span class="op">$</span>grouping
)</code></pre></div>
<p><img src="reference/figures/dimred-1.png" width="100%"></p>
<p>Similarly, the expression of a gene:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_dimred.html">plot_dimred</a></span>(
  model, 
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression,
  <span class="dt">feature_oi =</span> <span class="st">"Fn1"</span>
)</code></pre></div>
<p><img src="reference/figures/dimred_expression-1.png" width="100%"></p>
<p>Groups can also be visualised using a background color</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_dimred.html">plot_dimred</a></span>(
  model, 
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression, 
  <span class="dt">color_cells =</span> <span class="st">"feature"</span>,
  <span class="dt">feature_oi =</span> <span class="st">"Vim"</span>,
  <span class="dt">color_density =</span> <span class="st">"grouping"</span>,
  <span class="dt">grouping =</span> fibroblast_reprogramming_treutlein<span class="op">$</span>grouping,
  <span class="dt">label_milestones =</span> <span class="ot">FALSE</span>
)</code></pre></div>
<p><img src="reference/figures/dimred_groups-1.png" width="100%"></p>
</div>
<div id="predicting-and-visualising-genes-of-interest" class="section level3">
<h3 class="hasAnchor">
<a href="#predicting-and-visualising-genes-of-interest" class="anchor"></a>Predicting and visualising genes of interest</h3>
<p>We integrate several methods to extract candidate marker genes/features from a trajectory.</p>
<div id="a-global-overview-of-the-most-predictive-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#a-global-overview-of-the-most-predictive-genes" class="anchor"></a>A global overview of the most predictive genes</h4>
<p>At default, the overall most important genes are calculated when plotting a heatmap.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_heatmap.html">plot_heatmap</a></span>(
  model,
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression,
  <span class="dt">grouping =</span> fibroblast_reprogramming_treutlein<span class="op">$</span>grouping,
  <span class="dt">features_oi =</span> <span class="dv">50</span>
)</code></pre></div>
<p><img src="reference/figures/heatmap-1.png" width="100%"></p>
</div>
<div id="lineagebranch-markers" class="section level4">
<h4 class="hasAnchor">
<a href="#lineagebranch-markers" class="anchor"></a>Lineage/branch markers</h4>
<p>We can also extract features specific for a branch, eg. genes which change when a cell differentiates into a Neuron</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">branch_feature_importance &lt;-<span class="st"> </span><span class="kw"><a href="reference/calculate_overall_feature_importance.html">calculate_branch_feature_importance</a></span>(model, <span class="dt">expression_source=</span>dataset<span class="op">$</span>expression)

neuron_features &lt;-<span class="st"> </span>branch_feature_importance <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(to <span class="op">==</span><span class="st"> </span><span class="kw">which</span>(model<span class="op">$</span>milestone_labelling <span class="op">==</span><span class="st">"Neuron"</span>)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">top_n</span>(<span class="dv">50</span>, importance) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">pull</span>(feature_id)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="reference/plot_heatmap.html">plot_heatmap</a></span>(
  model, 
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression, 
  <span class="dt">features_oi =</span> neuron_features
)</code></pre></div>
<p><img src="reference/figures/branch-1.png" width="100%"></p>
</div>
<div id="genes-important-at-bifurcation-points" class="section level4">
<h4 class="hasAnchor">
<a href="#genes-important-at-bifurcation-points" class="anchor"></a>Genes important at bifurcation points</h4>
<p>We can also extract features which change at the branching point</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">branching_milestone &lt;-<span class="st"> </span>model<span class="op">$</span>milestone_network <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(from) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="kw">n</span>() <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(from) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">first</span>()

branch_feature_importance &lt;-<span class="st"> </span><span class="kw"><a href="reference/calculate_overall_feature_importance.html">calculate_branching_point_feature_importance</a></span>(model, <span class="dt">expression_source=</span>dataset<span class="op">$</span>expression, <span class="dt">milestones_oi =</span> branching_milestone)

branching_point_features &lt;-<span class="st"> </span>branch_feature_importance <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">top_n</span>(<span class="dv">20</span>, importance) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(feature_id)

<span class="kw"><a href="reference/plot_heatmap.html">plot_heatmap</a></span>(
  model,
  <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression,
  <span class="dt">features_oi =</span> branching_point_features
)</code></pre></div>
<p><img src="reference/figures/branching_point-1.png" width="100%"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">space &lt;-<span class="st"> </span>dyndimred<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/dyndimred/topics/dimred">dimred_mds</a></span>(dataset<span class="op">$</span>expression)
<span class="kw">map</span>(branching_point_features[<span class="dv">1</span><span class="op">:</span><span class="dv">12</span>], <span class="cf">function</span>(feature_oi) {
  <span class="kw"><a href="reference/plot_dimred.html">plot_dimred</a></span>(model, <span class="dt">dimred =</span> space, <span class="dt">expression_source =</span> dataset<span class="op">$</span>expression, <span class="dt">feature_oi =</span> feature_oi, <span class="dt">label_milestones =</span> <span class="ot">FALSE</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">"none"</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(feature_oi)
}) <span class="op">%&gt;%</span><span class="st"> </span>patchwork<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/patchwork/topics/wrap_plots">wrap_plots</a></span>()</code></pre></div>
<p><img src="reference/figures/branching_point_dimred-1.png" width="100%"></p>
</div>
</div>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install dyno from github using:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools")</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"dynverse/dyno"</span>)</code></pre></div>
<p>On Linux, you will need to install udunits and ImageMagick:</p>
<ul>
<li>Debian / Ubuntu / Linux Mint: <code>sudo apt-get install libudunits2-dev imagemagick</code>
</li>
<li>Fedora / CentOS / RHEL: <code>sudo dnf install udunits2-devel ImageMagick-c++-devel</code>
</li>
</ul>
<p><a href="https://docs.docker.com/install">Docker</a> has to be installed to run TI methods. See <a href="https://docs.docker.com/install" class="uri">https://docs.docker.com/install</a> for instructions.</p>
<p>You can test whether docker is correctly installed by running:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dynwrap<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/dynwrap/topics/test_docker_installation">test_docker_installation</a></span>(<span class="dt">detailed =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ✔ Docker is installed</span>
<span class="co">#&gt; ✔ Docker daemon is running</span>
<span class="co">#&gt; ✔ Docker is at correct version (&gt;1.0): 1.38</span>
<span class="co">#&gt; ✔ Docker is in linux mode</span>
<span class="co">#&gt; ✔ Docker can pull images</span>
<span class="co">#&gt; ✔ Docker can run image</span>
<span class="co">#&gt; ✔ Docker can mount temporary volumes</span>
<span class="co">#&gt; ✔ Docker test successful -----------------------------------------------------------------</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>This command will give helpful tips if some parts of the installation are missing.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/dynverse/dyno">https://​github.com/​dynverse/​dyno</a>
</li>
<li>Report a bug at <br><a href="https://github.com/dynverse/dyno/issues">https://​github.com/​dynverse/​dyno/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Robrecht Cannoodt <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3641-729X" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" height="16"></a> </li>
<li>Wouter Saelens <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7114-6248" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" height="16"></a> </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
