<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Inferring a trajectory inference using <a href='https://doi.org/10.1186/s12859-016-1175-6'>cellTree with gibbs</a> — ti_celltree_gibbs • dyno</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Inferring a trajectory inference using <a href='https://doi.org/10.1186/s12859-016-1175-6'>cellTree with gibbs</a> — ti_celltree_gibbs" />

<meta property="og:description" content="Will generate a trajectory using cellTree with gibbs. This method was wrapped inside a container." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dyno</a>
        <span class="label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dyno.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dynverse/dyno">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Inferring a trajectory inference using <a href='https://doi.org/10.1186/s12859-016-1175-6'>cellTree with gibbs</a></h1>
    <small class="dont-index">Source: <a href='https://github.com/dynverse/dyno/blob/master/R/ti_celltree.R'><code>R/ti_celltree.R</code></a>, <a href='https://github.com/dynverse/dyno/blob/master/R/ti_container.R'><code>R/ti_container.R</code></a></small>
    <div class="hidden name"><code>ti_celltree_gibbs.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Will generate a trajectory using <a href='https://doi.org/10.1186/s12859-016-1175-6'>cellTree with gibbs</a>. This method was wrapped inside a <a href='https://github.com/dynverse/dynmethods/tree/master/containers/celltree_gibbs'>container</a>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>ti_celltree_gibbs</span>(<span class='kw'>docker</span> <span class='kw'>=</span> <span class='st'>"stevedore"</span> <span class='kw'>%in%</span> <span class='fu'>rownames</span>(<span class='fu'>installed.packages</span>()),
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"Gibbs"</span>, <span class='kw'>sd_filter</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>absolute_width</span> <span class='kw'>=</span> <span class='fl'>0</span>,
  <span class='kw'>width_scale_factor</span> <span class='kw'>=</span> <span class='fl'>1.5</span>, <span class='kw'>outlier_tolerance_factor</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>rooting_method</span> <span class='kw'>=</span> <span class='st'>"null"</span>, <span class='kw'>num_topics</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>tot_iter</span> <span class='kw'>=</span> <span class='fl'>200</span>,
  <span class='kw'>tolerance</span> <span class='kw'>=</span> <span class='fl'>1e-05</span>)

<span class='fu'>ti_celltree_gibbs</span>(<span class='kw'>docker</span> <span class='kw'>=</span> <span class='st'>"stevedore"</span> <span class='kw'>%in%</span> <span class='fu'>rownames</span>(<span class='fu'>installed.packages</span>()),
  <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"Gibbs"</span>, <span class='kw'>sd_filter</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>absolute_width</span> <span class='kw'>=</span> <span class='fl'>0</span>,
  <span class='kw'>width_scale_factor</span> <span class='kw'>=</span> <span class='fl'>1.5</span>, <span class='kw'>outlier_tolerance_factor</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>rooting_method</span> <span class='kw'>=</span> <span class='st'>"null"</span>, <span class='kw'>num_topics</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>tot_iter</span> <span class='kw'>=</span> <span class='fl'>200</span>,
  <span class='kw'>tolerance</span> <span class='kw'>=</span> <span class='fl'>1e-05</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>docker</th>
      <td><p>Whether to use the docker container or the R wrapper</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>LDA inference method to use. Can be any unique prefix of ‘maptpx’, ‘Gibbs’ or ‘VEM’ (defaults to ‘maptpx’)</p></td>
    </tr>
    <tr>
      <th>sd_filter</th>
      <td><p>Standard-deviation threshold below which genes should be removed from the data.</p></td>
    </tr>
    <tr>
      <th>absolute_width</th>
      <td><p>Distance threshold below which a cell vertex is considered to be attached to a backbone vertex (see paper for more details).
By default, this threshold is computed dynamically, based on the distance distribution for each branch.</p></td>
    </tr>
    <tr>
      <th>width_scale_factor</th>
      <td><p>A scaling factor for the dynamically-computed distance threshold (ignored if absolute_width is provided).
Higher values will result in less branches in the backbone tree, while lower values might lead to a large number of backbone branches.</p></td>
    </tr>
    <tr>
      <th>outlier_tolerance_factor</th>
      <td><p>Proportion of vertices, out of the total number of vertices divided by the total number of branches,
that can be left at the end of the backbone tree-building algorithm.</p></td>
    </tr>
    <tr>
      <th>rooting_method</th>
      <td><p>Method used to root the backbone tree. Must be one of: ‘null’, ‘longest.path’, ‘center.start.group’ or ‘average.start.group’.
‘longest.path' picks one end of the longest shortest-path between two vertices.
'center.start.group’ picks the vertex in the starting group with lowest mean-square-distance to the others.
‘average.start.group’ creates a new artificial vertex, as the average of all cells in the starting group.
‘null’ picks the best method based on the type of grouping and start group information available.</p></td>
    </tr>
    <tr>
      <th>num_topics</th>
      <td><p>The number of topics to fit in the model.</p></td>
    </tr>
    <tr>
      <th>tot_iter</th>
      <td><p>Number of iterations of the LDA inference.</p></td>
    </tr>
    <tr>
      <th>tolerance</th>
      <td><p>Tolerance values of the LDA inference.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The trajectory model</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This methods was first wrapped inside R, see ti_celltree_gibbs</p>
<p>The method is described in: <a href='https://doi.org/10.1186/s12859-016-1175-6'>duVerle, D.A., Yotsukura, S., Nomura, S., Aburatani, H., Tsuda, K., 2016. CellTree: an R/bioconductor package to infer the hierarchical structure of cell populations from single-cell RNA-seq data. BMC Bioinformatics 17.</a></p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

