<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Inferring a trajectory inference using RaceID / StemID — ti_raceid_stemid • dyno</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Inferring a trajectory inference using RaceID / StemID — ti_raceid_stemid" />

<meta property="og:description" content="Will generate a trajectory using RaceID / StemID.
This method was wrapped inside a
container.
The original code of this method is available
here." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dyno</a>
        <span class="label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dyno.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dynverse/dyno">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Inferring a trajectory inference using RaceID / StemID</h1>
    <small class="dont-index">Source: <a href='https://github.com/dynverse/dyno/blob/master/R/ti_raceid_stemid.R'><code>R/ti_raceid_stemid.R</code></a></small>
    <div class="hidden name"><code>ti_raceid_stemid.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Will generate a trajectory using <a href='https://doi.org/10.1016/j.stem.2016.05.010'>RaceID / StemID</a>.</p>
<p>This method was wrapped inside a
<a href='https://github.com/dynverse/dynmethods/tree/master/containers/raceid_stemid'>container</a>.
The original code of this method is available
<a href='https://github.com/dgrun/RaceID3_StemID2_package'>here</a>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>ti_raceid_stemid</span>(<span class='kw'>knn</span> <span class='kw'>=</span> <span class='fl'>10L</span>, <span class='kw'>ccor</span> <span class='kw'>=</span> <span class='fl'>0.4</span>, <span class='kw'>metric</span> <span class='kw'>=</span> <span class='st'>"pearson"</span>,
  <span class='kw'>sat</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>samp</span> <span class='kw'>=</span> <span class='fl'>1000L</span>, <span class='kw'>cln</span> <span class='kw'>=</span> <span class='fl'>30L</span>, <span class='kw'>clustnr</span> <span class='kw'>=</span> <span class='fl'>30L</span>,
  <span class='kw'>bootnr</span> <span class='kw'>=</span> <span class='fl'>50L</span>, <span class='kw'>FUNcluster</span> <span class='kw'>=</span> <span class='st'>"kmedoids"</span>, <span class='kw'>probthr</span> <span class='kw'>=</span> <span class='fl'>0.001</span>,
  <span class='kw'>outminc</span> <span class='kw'>=</span> <span class='fl'>5L</span>, <span class='kw'>outlg</span> <span class='kw'>=</span> <span class='fl'>2L</span>, <span class='kw'>outdistquant</span> <span class='kw'>=</span> <span class='fl'>0.95</span>,
  <span class='kw'>initial_cmd</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>perplexity</span> <span class='kw'>=</span> <span class='fl'>30L</span>, <span class='kw'>cthr</span> <span class='kw'>=</span> <span class='fl'>5L</span>, <span class='kw'>nmode</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>projcells_knn</span> <span class='kw'>=</span> <span class='fl'>3L</span>, <span class='kw'>fr</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>pdishuf</span> <span class='kw'>=</span> <span class='fl'>500L</span>, <span class='kw'>fast</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>pthr</span> <span class='kw'>=</span> <span class='fl'>0.01</span>, <span class='kw'>scthr</span> <span class='kw'>=</span> <span class='fl'>0.2</span>, <span class='kw'>run_environment</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>knn</th>
      <td><p>integer; Number of nearest neighbors used to infer corresponding
cell types in different batches. (default: <code>10L</code>; range: from <code>5L</code> to <code>50L</code>)</p></td>
    </tr>
    <tr>
      <th>ccor</th>
      <td><p>numeric; Correlation coefficient used as a treshhold for
determining genes correlated to eachother. (default: <code>0.4</code>; range: from <code>0</code> to
<code>1</code>)</p></td>
    </tr>
    <tr>
      <th>metric</th>
      <td><p>discrete; Distances are computed from the filtered expression
matrix after optional feature selection, dimensional reduction, and/or
transformation (batch correction). (default: <code>"pearson"</code>; values: <code>"pearson"</code>,
<code>"spearman"</code>, <code>"logpearson"</code>, <code>"euclidean"</code>)</p></td>
    </tr>
    <tr>
      <th>sat</th>
      <td><p>logical; If <code>TRUE</code>, then the number of clusters is determined
based on finding the saturation point of the mean within-cluster dispersion as
a function of the cluster number. If <code>FALSE</code>, then cluster number needs to
be given as <code>cln</code>.</p></td>
    </tr>
    <tr>
      <th>samp</th>
      <td><p>integer; Number of booststrapping runs for clusterboot. Default is
50 (default: <code>1000L</code>; range: from <code>50L</code> to <code>10000L</code>)</p></td>
    </tr>
    <tr>
      <th>cln</th>
      <td><p>integer; Number of clusters to be used. If <code>sat</code> is
<code>TRUE</code>, this number is inferred by the saturation criterion. (default:
<code>30L</code>; range: from <code>10L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>clustnr</th>
      <td><p>integer; Maximum number of clusters for the derivation of the
cluster number by the saturation of mean within-cluster-dispersion. (default:
<code>30L</code>; range: from <code>10L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>bootnr</th>
      <td><p>integer; Number of booststrapping runs for clusterboot. (default:
<code>50L</code>; range: from <code>20L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>FUNcluster</th>
      <td><p>discrete; Clustering method used by RaceID3. (default:
<code>"kmedoids"</code>; values: <code>"kmedoids"</code>, <code>"kmeans"</code>, <code>"hclust"</code>)</p></td>
    </tr>
    <tr>
      <th>probthr</th>
      <td><p>numeric; Outlier probability threshold for a minimum of outlg
genes to be an outlier cell. This probability is computed from a negative
binomial background model of expression in a cluster. (default: <code>0.001</code>; range:
from <code>1e-05</code> to <code>1L</code>)</p></td>
    </tr>
    <tr>
      <th>outminc</th>
      <td><p>integer; Minimal transcript count of a gene in a clusters to be
tested for being an outlier gene. (default: <code>5L</code>; range: from <code>0L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>outlg</th>
      <td><p>integer; Minimum number of outlier genes required for being an
outlier cell. (default: <code>2L</code>; range: from <code>0L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>outdistquant</th>
      <td><p>numeric; Real number between zero and one. Outlier cells
are merged to outlier clusters if their distance smaller than the
outdistquant-quantile of the distance distribution of pairs of cells in the
orginal clusters after outlier removal. (default: <code>0.95</code>; range: from <code>0</code> to
<code>1</code>)</p></td>
    </tr>
    <tr>
      <th>initial_cmd</th>
      <td><p>logical; If TRUE, then the t-SNE map computation is
initialized with a configuration obtained by classical multidimensional
scaling.</p></td>
    </tr>
    <tr>
      <th>perplexity</th>
      <td><p>integer; Perplexity of the t-SNE map. (default: <code>30L</code>; range:
from <code>5L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>cthr</th>
      <td><p>integer; Clusters to be included into the StemID2 analysis must
contain more than cthr cells. D (default: <code>5L</code>; range: from <code>1L</code> to <code>25L</code>)</p></td>
    </tr>
    <tr>
      <th>nmode</th>
      <td><p>logical; If <code>TRUE</code>, then a cell of given cluster is assigned
to the link to the cluster with the smallest average distance of the knn
nearest neighbours within this cluster.</p></td>
    </tr>
    <tr>
      <th>projcells_knn</th>
      <td><p>integer; See <code>nmode</code>. (default: <code>3L</code>; range: from
<code>3L</code> to <code>20L</code>)</p></td>
    </tr>
    <tr>
      <th>fr</th>
      <td><p>logical; Use Fruchterman-Rheingold layout instead of t-SNE for
dimensional-reduction representation of the lineage tree.</p></td>
    </tr>
    <tr>
      <th>pdishuf</th>
      <td><p>integer; Number of randomizations of cell positions for which to
compute projections of cells on inter-cluster links. (default: <code>500L</code>; range:
from <code>10L</code> to <code>10000L</code>)</p></td>
    </tr>
    <tr>
      <th>fast</th>
      <td><p>logical; If <code>TRUE</code> and <code>nmode</code> is <code>FALSE</code> cells will
still be assigned to links based on maximum projections but a fast approximate
background model will be used to infer significance. The function will do
nothing in this case.</p></td>
    </tr>
    <tr>
      <th>pthr</th>
      <td><p>numeric; P-value cutoff for link significance. This threshold is
applied for the calculation of link scores reflecting how uniformly a link is
occupied by cells. (default: <code>0.01</code>; range: from <code>1e-05</code> to <code>1L</code>)</p></td>
    </tr>
    <tr>
      <th>scthr</th>
      <td><p>numeric; Score threshold for links to be shown in the graph.
(default: <code>0.2</code>; range: from <code>0</code> to <code>1</code>)</p></td>
    </tr>
    <tr>
      <th>run_environment</th>
      <td><p>In which environment to run the method, can be <code>"docker"</code> or <code>"singularity"</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A TI method wrapper to be used together with
<code><a href='http://www.rdocumentation.org/packages/dynwrap/topics/infer_trajectories'>infer_trajectory</a></code></p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Grün, D., Muraro, M.J., Boisset, J.-C., Wiebrands, K., Lyubimova,
A., Dharmadhikari, G., van den Born, M., van Es, J., Jansen, E., Clevers, H.,
de Koning, E.J.P., van Oudenaarden, A., 2016. De Novo Prediction of Stem Cell
Identity using Single-Cell Transcriptome Data. Cell Stem Cell 19, 266–277.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#references">References</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

