<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Inferring a trajectory inference using SCUBA — ti_scuba • dyno</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Inferring a trajectory inference using SCUBA — ti_scuba" />

<meta property="og:description" content="Will generate a trajectory using
SCUBA.
This method was wrapped inside a
container.
The original code of this method is available
here." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dyno</a>
        <span class="label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dyno.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dynverse/dyno">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Inferring a trajectory inference using SCUBA</h1>
    <small class="dont-index">Source: <a href='https://github.com/dynverse/dyno/blob/master/R/ti_scuba.R'><code>R/ti_scuba.R</code></a></small>
    <div class="hidden name"><code>ti_scuba.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Will generate a trajectory using
<a href='https://doi.org/10.1073/pnas.1408993111'>SCUBA</a>.</p>
<p>This method was wrapped inside a
<a href='https://github.com/dynverse/dynmethods/tree/master/containers/scuba'>container</a>.
The original code of this method is available
<a href='https://github.com/gcyuan/SCUBA'>here</a>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>ti_scuba</span>(<span class='kw'>rigorous_gap_stats</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>N_dim</span> <span class='kw'>=</span> <span class='fl'>2L</span>,
  <span class='kw'>low_gene_threshold</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>low_gene_fraction_max</span> <span class='kw'>=</span> <span class='fl'>0.7</span>,
  <span class='kw'>min_split</span> <span class='kw'>=</span> <span class='fl'>15L</span>, <span class='kw'>min_percentage_split</span> <span class='kw'>=</span> <span class='fl'>0.25</span>,
  <span class='kw'>run_environment</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>rigorous_gap_stats</th>
      <td><p>logical; Whether to use rigorous gap statistics to
determine number of clusters</p></td>
    </tr>
    <tr>
      <th>N_dim</th>
      <td><p>integer; Number of TSNE dimensions (default: <code>2L</code>; range: from
<code>2L</code> to <code>3L</code>)</p></td>
    </tr>
    <tr>
      <th>low_gene_threshold</th>
      <td><p>numeric; Threshold value for genes of low expression
levels (default: <code>1L</code>; range: from <code>0L</code> to <code>5L</code>)</p></td>
    </tr>
    <tr>
      <th>low_gene_fraction_max</th>
      <td><p>numeric; Maximum fraction of lowly-expressed cells
allowed for each gene (default: <code>0.7</code>; range: from <code>0L</code> to <code>1L</code>)</p></td>
    </tr>
    <tr>
      <th>min_split</th>
      <td><p>integer; Lower threshold on the number of cells in a cluster
for this cluster to be split. (default: <code>15L</code>; range: from <code>1L</code> to <code>100L</code>)</p></td>
    </tr>
    <tr>
      <th>min_percentage_split</th>
      <td><p>numeric; Minimum fraction of cells in the smaller
cluster during a bifurcation. (default: <code>0.25</code>; range: from <code>0L</code> to <code>1L</code>)</p></td>
    </tr>
    <tr>
      <th>run_environment</th>
      <td><p>In which environment to run the method, can be <code>"docker"</code> or <code>"singularity"</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A TI method wrapper to be used together with
<code><a href='http://www.rdocumentation.org/packages/dynwrap/topics/infer_trajectories'>infer_trajectory</a></code></p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Marco, E., Karp, R.L., Guo, G., Robson, P., Hart, A.H., Trippa, L.,
Yuan, G.-C., 2014. Bifurcation analysis of single-cell gene expression data
reveals epigenetic landscape. Proceedings of the National Academy of Sciences
111, E5643–E5650.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#references">References</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Robrecht Cannoodt, Wouter Saelens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

